### ISSUE TITLE
Ratchet On Desktop Questions

### ISSUE THREAD

**Issue by geggleto** [twbs/ratchet#636]
Hi,

I've been poking around and developing some small web apps with Ratchet (which is freaking awesome)... And I was asked to create a desktop version using the same sort of UI.

I read about `fingerbang.js` and emulating touch events... but I came across another post and read the source for Ratchet... It just seems that you disable NORMAL click events...

To enable Ratchet on Desktop is it as simple as...

```
  window.addEventListener('click', function (e) { if (getTarget(e)) {e.preventDefault();} });
```

to

```
 window.addEventListener('click', touchend);
```

Now I ask this because I am not sure why the design choice was made to make the project mobile only. It would be nice to know at least for me the certain restrictions around the framework and what features might or might not work.

Thank you so much for your work.

**Comment by isikfsc** [comment]
Hi I have found the [react-ratchet](http://react-components.com/component/react-ratchet) on react-components and the example product they provide is [hacker news mobile](http://hackernewsmobile.com).
It works with its links on my Desktop Safari web browser as opposed to several issues on ratchet clicks not working on desktop.
How they get this work on desktop.
There are some other frameworks without this restriction but they miss the points Ratchet gets very right, yet Ratchet feels nicer on Desktop than lots of them. Even Bootstrap.
Is that possible by simple steps like mentioned in the first post in this isssue thread.

**Comment by geggleto** [comment]
Yeah, I was able to enable desktop usage by simply disabling PUSH links, but ,my app is not very large.
The method I posted above should enable it as well.

**Comment by isikfsc** [comment]
Will this disable usage for mobile clients? If yes will a conditional work to have both desktop and mobile clients work without problems from same codebase?

**Comment by geggleto** [comment]
either solution does not break mobile

**Comment by isikfsc** [comment]
thanks. Now you mean it is ok to go with Ratchet in production with this custom configuration.
Something like:
`
if(<client is mobile>)
    window.addEventListener('click', function (e) { if (getTarget(e)) {e.preventDefault();} });
else if(<client is desktop>)
    window.addEventListener('click', touchend);
`

**Comment by geggleto** [comment]
Nope...
```
  // Attach PUSH event handlers
  // ==========================

  window.addEventListener('touchstart', function () { isScrolling = false; });
  window.addEventListener('touchmove', function () { isScrolling = true; });
  window.addEventListener('touchend', touchend);
  window.addEventListener('click', /*function (e) { if (getTarget(e)) {e.preventDefault();} }*/ touchend); ///<--
  window.addEventListener('popstate', popstate);
  window.PUSH = PUSH;

```

**Comment by isikfsc** [comment]
Not works?
Is this the all modification needed?

I am trying the ratchet examples in the directory.
I have changed the ratchet.js in the list directory and linked all example index.html files to this ratchet.js instead of ratchet.min.js.
